src/components/SendSykmelding/FormSections/Arbeidssituasjon/ArbeidssituasjonSection.tsx

```diff
@@ -81,10 +93,19 @@ function useDynamicSubSections(
         arbeidssituasjon === ArbeidssituasjonType.NAERINGSDRIVENDE
     const shouldShowEgenmeldingsperioderSporsmal: boolean =
         isFrilanserOrNaeringsdrivende && !sykmeldingUtenforVentetid.erUtenforVentetid
-    const shouldShowSendesTilArbeidsgiverInfo =
+    const hasEgenmeldingsperioderAnsatt: boolean | null =
+        egenmeldingsperioderAnsatt && egenmeldingsperioderAnsatt.length >= 1
+    const hasCompletedEgenmeldingsperioderAnsatt: boolean =
+        (hasEgenmeldingsperioderAnsatt &&
+            egenmeldingsperioderAnsatt?.[egenmeldingsperioderAnsatt.length - 1].harPerioder === YesOrNo.NO) ||
+        (hasEgenmeldingsperioderAnsatt &&
+            egenmeldingsperioderAnsatt?.[egenmeldingsperioderAnsatt.length - 1].hasClickedVidere === true) ||
+        arbeidssituasjon !== ArbeidssituasjonType.ARBEIDSTAKER
```

Comment:
denne logikken er veldig tung. :weary: burde kanskje delt dette opp i litt mindre funksjoner? Mye and/or og nullabilitet på en plass. :stuck_out_tongue: